### DISCUSSION TITLE
How to use Nimbro now that "nimbro.launch" was deleted

### DISCUSSION THREAD

**Discussion by MartinSegaert** [ctu-mrs/mrs_uav_system#155] [Q&A] [answered]
Hello,
Some month ago we used nimbro network to do some collision avoidance between UAV's. At that time, it worked okay.
In the session.yml we used the following:

`- nimbro:
      layout: tiled
      panes:
        - waitForRos; roslaunch mrs_uav_general nimbro.launch custom_config_uav_names:=custom_configs/uav_names.yaml`

which contains "nimbro.launch".

Now, I want to use nimbro for some other application. I tested nimbro again for the collision avoidance some weeks ago (files were not modified for a long time), but it did not work. Now I just saw that the file "nimbro.launch" was deleted in the mrs_workspace in octobre 2022.
 
![Screenshot from 2023-02-14 15-41-47](https://user-images.githubusercontent.com/62548016/218993342-54258c1f-6f75-459c-9ce0-79414242cbc9.png)

My question is: How should I adapt my files/codes to use nimbro like before, now that "nimbro.launch" does not exist anymore ?

Thanks for your help.

Martin

**Comment by klaxalk** [comment]
Hey, we are now running nimbro using this script: https://github.com/ctu-mrs/mrs_uav_general/blob/master/scripts/run_nimbro.py

**Comment by klaxalk** [comment, marked as answer]
And we run it now as

```bash
waitForRos; rosrun mrs_uav_general run_nimbro.py custom_configs/nimbro.yaml custom_configs/uav_names.yaml
```

**Reply by MartinSegaert** [reply]
Thanks for your response.
 
Also, were can I find "nimbro.yaml" to see how it works ? What should be in that file ?

**Reply by MartinSegaert** [reply]
@klaxalk 
I'm sorry to bother you by asking again.
If possible I would like to resolve this issue this week.

Thanks in advance,

Martin

**Reply by klaxalk** [reply]
Sorry, I missed your question. The full session example that goes with Nimbro is here: [https://github.com/ctu-mrs/uav_core/tree/master/tmux_scripts/swarming_template](https://github.com/ctu-mrs/uav_core/tree/master/tmux_scripts/swarming_template)

Alternatively, the default config for Nimbro is here: [https://github.com/ctu-mrs/mrs_uav_general/blob/master/config/communication_config.yaml](https://github.com/ctu-mrs/mrs_uav_general/blob/master/config/communication_config.yaml), and the default config for the uav_names.list is here [https://github.com/ctu-mrs/mrs_uav_general/blob/master/config/uav_names.yaml](https://github.com/ctu-mrs/mrs_uav_general/blob/master/config/uav_names.yaml)

**Reply by MartinSegaert** [reply]
@klaxalk 
Thanks for your quick response. I tried to make it work this morning. "run_nimbro.py" gave me some errors, which at first I was able to solve. 
But now, I have an error I can't solve:

`ERROR: cannot launch node of type [nimbro_topic_transport/sender]: Cannot locate node of type [sender] in package [nimbro_topic_transport]. Make sure file exists in package path and permission is set to executable (chmod +x)`

I have the package "nimbro_network" in my workspace, which contains "nimbro_topic_transport", but I can't see any "Sender".

Do you have any idea what I can do to solve this issue ?

**Reply by spurnvoj** [reply]
Hi @MartinSegaert , check that you have the up-to-date version of `uav_modules` https://github.com/ctu-mrs/uav_modules.

**Reply by MartinSegaert** [reply]
Hi @spurnvoj,

This Wednesday I did a fresh install of the "mrs_uav_system" and the system of our own research department which additionally installs the up-to-date versions of "mrs_gazebo_extras_resources", "trajectory_loader", "mrs_serial" and "nimbro_network", But not the whole "uav_modules". 

Do I need the whole uav_modules ? If not, the required modules are up-to-date as I installed them Wednesday.
Furthermore, the build status of uav_modules is failing.

Thanks again for your help so far,

martin

**Reply by spurnvoj** [reply]
Hi @MartinSegaert, we are using `develop` branch for the nimbro_network repository (see https://github.com/ctu-mrs/uav_modules/blob/master/.gitman.yml#L3-L5). The default branch of this repo was `master` until now. I just changed it to `develop` to be sure that people will use the right branch if they clone it separately, as you did. So please checkout the `develop` branch and let me know if it works.

**Reply by MartinSegaert** [reply]
Hi @spurnvoj ,
Now it works !!
I simply did a checkout to the develop branch in the folder nimbro_network as you said, and it works fine.

Thanks a lot for your help @spurnvoj and @klaxalk .
